### ISSUE TITLE
[Fix Provided] Excessive logging from sp_save_long_queries() -  SPRO Save Profile failed: Out of range value for column 'duration' at row 1

### ISSUE THREAD

**Issue by gosixl** [superspeedyplugins/issue-tracker#139]
Plugin: Scalability Pro / Occurrence: Constantly / Version: 5.52

In sp_save_long_queries() there seems to be a bug whereby the global $start_time comes out as null. This then results in many debug logs per second of `[06-Apr-2024 13:34:50 UTC] SPRO Save Profile failed: Out of range value for column 'duration' at row 1`. This is because `$duration` will be: `$end_time` (now) - null

I solved this by adding  `if (!$start_time) return;` under line 23 (the global declaration of `$start_time` and `$wpdb`

The stack trace says all of these instances come from woocommerce's `get_patterns_ai_data_post()` in `/woocommerce/src/Blocks/AIContent/PatternsHelper.php`

```
Stack trace:
    0 wp-includes/class-wp-query.php(3373): apply_filters_ref_array()\n#
    1 wp-content/plugins/woocommerce/src/Blocks/AIContent/PatternsHelper.php(47): WP_Query->get_posts()\n#
    2 wp-content/plugins/woocommerce/src/Blocks/AIContent/PatternsHelper.php(109): Automattic\WooCommerce\Blocks\AIContent\PatternsHelper::get_patterns_ai_data_post()\n#
    3 wp-content/plugins/woocommerce/src/Blocks/BlockPatterns.php(133): Automattic\WooCommerce\Blocks\AIContent\PatternsHelper::get_patterns_dictionary()\n#
    4 wp-includes/class-wp-hook.php(324): Automattic\WooCommerce\Blocks\BlockPatterns->register_block_patterns()\n#
    5 wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()\n#
    6 wp-includes/plugin.php(517): WP_Hook->do_action()\n#
    7 wp-settings.php(695): do_action()\n#
    8 wp-config.php(69): require_once()\n#
    9 wp-load.php(50): require_once()\n#
    10 wp-blog-header.php(13): require_once()\n#
    11 index.php(17): require()\n", 
```

And this is the SQL query for what its worth:     `"SELECT a.*, g.slug AS group FROM wp_actionscheduler_actions a LEFT JOIN wp_actionscheduler_groups g ON a.group_id=g.group_id WHERE a.action_id=3715129"`

**Comment by gosixl** [comment]
Hi guys.. would someone please be able to advise here. I haven't updated my plugin since to avoid reintroducing this issue

**Comment by dhilditch** [comment]
@gosixl I'll take a look at this myself

**Comment by gosixl** [comment]
Much appreciated Dave. Hopefully my proposed fix gets us halfway there 👍

**Comment by dhilditch** [comment]
Fixed in 5.60

**Comment by gosixl** [comment]
Appreciate it 👍
